hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientClusterService.java

```diff
@@ -114,4 +114,9 @@
      */
     Address getOwnerConnectionAddress();
 
+    /**
+     *
+     * @return The connected server hazelcast version as calculated by BuilInfo class.
+     */
+    int getHazelcastServerVersion();
```

Comment:
IMHO instead of exposing `Connection` in `ClientInvocationService`, I would rather keep `ClientClusterService.getClusterVersion()` as I think it more accurately describes it's intended meaning: we want to decide which type to use based on the cluster version.
In any case, since there is currently no notion of cluster version, we could as well defer this change (if needed) to a further PR in the context of rolling-upgrades, I am just worried that if we do introduce a new method in SPI in this PR we may be stuck with it. wdyt?